include/mbedtls/ssl_internal.h

```diff
@@ -311,8 +311,15 @@ struct mbedtls_ssl_handshake_params
         ssl_ecrs_cke_ecdh_calc_secret,  /*!< ClientKeyExchange: ECDH step 2 */
         ssl_ecrs_crt_vrfy_sign,         /*!< CertificateVerify: pk_sign()   */
     } ecrs_state;                       /*!< current (or last) operation    */
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+    mbedtls_x509_crt *ecrs_peer_cert;        /*!< The peer's CRT chain.     */
```

Comment:
That looks like a bug - in a PSK-only configuration with restartable ECC enabled (which doesn't make any sense, but it's possible), the library would fail to build. I will fix it - thanks for spotting!

Reply:
Actually, it's not a bug, but rather by luck: `MBEDTLS_SSL__ECP_RESTARTABLE` is set only if `MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED` is enabled, which requires `MBEDTLS_X509_CRT_PARSE_C` to be set.

Reply:
I think it makes sense to remove the `MBEDTLS_X509_CRT_PARSE_C` guard here, then.

Reply:
Removed.